src/utils/googleTagManager.ts

```diff
@@ -155,14 +165,57 @@ export const trackEvent = ({
   action: string
   label?: EventLabel
 }): void => {
-  const dataLayer = {
+  const dataLayer: EventDataLayer = {
     event,
     chainId: _getChainId(),
     eventCategory: category,
     eventAction: action,
     eventLabel: tryParse(label),
   }
 
+  track(dataLayer)
+}
+
+type SafeAppEventDataLayer = {
+  event: GTM_EVENT
+  chainId: string
+  safeAppName: string
+  safeAppMethod: string
+  safeAppParams?: string
+  safeAppEthMethod?: string
+  safeAppSDKVersion?: string
+}
+
+export const getSafeAppName = (safeApp?: SafeApp): string => {
+  const parsedSafeApp = JSON.parse(safeApp?.id || 'Unknown Safe App')
+
+  return parsedSafeApp.name || parsedSafeApp.url || parsedSafeApp
+}
+
+export const trackSafeAppMessage = ({
+  app,
+  method,
+  params,
+  sdkVersion,
+}: {
+  app?: SafeApp
+  method: string
+  params?: any
```

Comment:
Do you not have a type for this? I'm surprise the linter hasn't picked up the usage of `any`.

Reply:
nope, it can be any type

Reply:
How so? Surely we have a required set of params? Or am I missing something?

Reply:
it can be any thing coming from the SDK https://github.com/gnosis/safe-apps-sdk/blob/8c1d1e4ece68e35dba6027f7adc6d31ad6b5caea/packages/safe-apps-provider/dist/provider.d.ts#L15

Reply:
Sometimes undefined, sometimes an object with custom data, others an object with arrays inside with object items :D

Reply:
![image](https://user-images.githubusercontent.com/1576776/162418422-850d2856-0a4f-4dac-a977-561dcb321a51.png)

Reply:
Crazy. I'd suggest being as strict as possible, maybe opting for: `Record<string, unknown> | undefined` perhaps?

Reply:
I think is fine like this aaron. Is the same we have in the SDK. It can be really any

Reply:
I know I'm being picky but the linter will complain if you check it.

Reply:
mmm, but the eslint action passes and no error in my terminal. I guess is running eslint behind the scenes

Reply:
I'm a sucker for strict typing but we can leave it as is.

Reply:
To run the linter it's `yarn lint:check`.

Reply:
ah ok a warning. I don't know what to say. Is really any :D

Reply:
`unknown` would have probably worked too.

Reply:
Yep, but in that case I have to cast to access properties 🙃